### DISCUSSION TITLE
Postfix admin hacking attempts

### DISCUSSION THREAD

**Discussion by 2mstorm** [postfixadmin/postfixadmin#695] [General]
Hello,
I configured postfixadmin on Debian 11 and it works great except I have noticed a lot of entries in my mysql-query.log like this

`2022-12-12T18:06:18        1533 Query     SELECT username AS user,password FROM mailbox WHERE username = 'root@mydomain.com' AND active='1'`
`2022-12-12T18:07:04         1533 Query     SELECT username AS user,password FROM mailbox WHERE username = 'sysadmin@mydomain.com' AND active='1'`
`2022-12-12T18:07:17         1533 Query     SELECT username AS user,password FROM mailbox WHERE username = 'admin2@mydomain.com' AND active='1'`

It looks like someone is probing the system because those usernames don't exist but when I check the log files for postfix admin (/var/log/apache2/postfixadmin_error.log and access log) it only shows my personal logins and IP.
I'm trying to understand what page (if any) they are using to probe the database and how would I find their IP
Thanks

**Comment by DavidGoodwin** [comment]
If there's nothing in Apache, then I suspect what you're seeing is Postfix doing lookups. 

Try looking in /var/log/mail.log ?

**Comment by 2mstorm** [comment]
I dug deeper - I have postfix, postfixAdmin, Dovecot and Roundcube all setup together on the system and the only other entries I found were in dovecot.log. So while it looks like postfixAdmin user is looking up the mysql database, (Connect   postfixadmin@localhost on postfixadmin using Socket) it seems like it originates from dovecot. Probably a spammer trying to find a way to exploit my mail server

`
mysql_query.log

2022-12-13T14:28:26.821210-08:00         2079 Connect   postfixadmin@localhost on postfixadmin using Socket
2022-12-13T14:28:26.824564-08:00         2079 Query     SELECT username AS user,password FROM mailbox WHERE username = 'eric@mydomain.com' AND active='1'
2022-12-13T14:28:28.243646-08:00         2079 Query     SELECT username AS user,password FROM mailbox WHERE username = 'manager@mydomain.com' AND active='1'
2022-12-13T14:29:28.304344-08:00         2079 Quit
2022-12-13T14:29:36.671223-08:00         2080 Connect   postfixadmin@localhost on postfixadmin using Socket
2022-12-13T14:29:36.672271-08:00         2080 Query     SELECT username AS user,password FROM mailbox WHERE username = 'alan@mydomain.com' AND active='1'
2022-12-13T14:30:11.778872-08:00         2080 Query     SELECT username AS user,password FROM mailbox WHERE username = 'xerox@mydomain.com' AND active='1'
2022-12-13T14:31:11.829703-08:00         2080 Quit
2022-12-13T14:32:54.980039-08:00         2081 Connect   postfixadmin@localhost on postfixadmin using Socket
2022-12-13T14:32:54.983318-08:00         2081 Query     SELECT username AS user,password FROM mailbox WHERE username = 'noreply@mydomain.com' AND active='1'
2022-12-13T14:33:05.899035-08:00         2081 Query     SELECT username AS user,password FROM mailbox WHERE username = 'kodi@mydomain.com' AND active='1'
2022-12-13T14:33:18.627390-08:00         2081 Query     SELECT username AS user,password FROM mailbox WHERE username = 'admin2@mydomain.com' AND active='1'
2022-12-13T14:34:18.689240-08:00         2081 Quit

dovecot.log

Dec 13 14:28:26 auth: Debug: Loading modules from directory: /usr/lib/dovecot/modules/auth
Dec 13 14:28:26 auth: Debug: Module loaded: /usr/lib/dovecot/modules/auth/lib20_auth_var_expand_crypt.so
Dec 13 14:28:26 auth: Debug: Module loaded: /usr/lib/dovecot/modules/auth/libdriver_mysql.so
Dec 13 14:28:26 auth: Debug: Read auth token secret from /run/dovecot/auth-token-secret.dat
Dec 13 14:28:26 auth: Debug: auth client connected (pid=0)
Dec 13 14:28:26 auth: Debug: client in: AUTH    1       LOGIN   service=smtp    nologin lip=66.228.62.176       rip=141.98.11.65        secured
Dec 13 14:28:26 auth: Debug: client passdb out: CONT    1       VXNlcm5hbWU6
Dec 13 14:28:26 auth: Debug: client in: CONT    1       ZXJpYw== (previous base64 data may contain sensitive data)
Dec 13 14:28:26 auth: Debug: client passdb out: CONT    1       UGFzc3dvcmQ6
Dec 13 14:28:26 auth: Debug: client in: CONT    1       ZXJpYyFAIyQl (previous base64 data may contain sensitive data)
Dec 13 14:28:26 auth: Debug: sql(eric@mydomain.com,141.98.11.65): Performing passdb lookup
Dec 13 14:28:26 auth-worker(30843): Debug: Loading modules from directory: /usr/lib/dovecot/modules/auth
Dec 13 14:28:26 auth-worker(30843): Debug: Module loaded: /usr/lib/dovecot/modules/auth/lib20_auth_var_expand_crypt.so
Dec 13 14:28:26 auth-worker(30843): Debug: Module loaded: /usr/lib/dovecot/modules/auth/libdriver_mysql.so
Dec 13 14:28:26 auth-worker(30843): Debug: conn unix:auth-worker (pid=30842,uid=109): Server accepted connection (fd=14)
Dec 13 14:28:26 auth-worker(30843): Debug: conn unix:auth-worker (pid=30842,uid=109): Sending version handshake
Dec 13 14:28:26 auth-worker(30843): Debug: conn unix:auth-worker (pid=30842,uid=109): auth-worker<1>: Handling PASSV request
Dec 13 14:28:26 auth-worker(30843): Debug: conn unix:auth-worker (pid=30842,uid=109): auth-worker<1>: sql(eric@mydomain.com,141.98.11.65): Performing passdb lookup
Dec 13 14:28:26 auth-worker(30843): Debug: conn unix:auth-worker (pid=30842,uid=109): auth-worker<1>: sql(eric@mydomain.com,141.98.11.65): query: SELECT username AS user,password FROM mailbox WHERE username = 'eric@mydomain.com' AND acti
ve='1'
Dec 13 14:28:26 auth-worker(30843): Info: conn unix:auth-worker (pid=30842,uid=109): auth-worker<1>: sql(eric@mydomain.com,141.98.11.65): unknown user
Dec 13 14:28:26 auth: Debug: sql(eric@mydomain.com,141.98.11.65): Finished passdb lookup
Dec 13 14:28:26 auth: Debug: auth(eric@mydomain.com,141.98.11.65): Auth request finished
Dec 13 14:28:26 auth-worker(30843): Debug: conn unix:auth-worker (pid=30842,uid=109): auth-worker<1>: sql(eric@mydomain.com,141.98.11.65): Finished passdb lookup
Dec 13 14:28:26 auth-worker(30843): Debug: conn unix:auth-worker (pid=30842,uid=109): auth-worker<1>: Finished
Dec 13 14:28:27 auth: Debug: auth client connected (pid=0)
Dec 13 14:28:28 auth: Debug: client in: AUTH    1       LOGIN   service=smtp    nologin lip=66.228.62.176       rip=45.125.66.22        secured
Dec 13 14:28:28 auth: Debug: client passdb out: CONT    1       VXNlcm5hbWU6
Dec 13 14:28:28 auth: Debug: client in: CONT    1       bWFuYWdlcg== (previous base64 data may contain sensitive data)
Dec 13 14:28:28 auth: Debug: client passdb out: CONT    1       UGFzc3dvcmQ6
Dec 13 14:28:28 auth: Debug: client in: CONT    1       bWFuYWdlciFAIyQl (previous base64 data may contain sensitive data)
Dec 13 14:28:28 auth: Debug: sql(manager@mydomain.com,45.125.66.22): Performing passdb lookup
Dec 13 14:28:28 auth-worker(30843): Debug: conn unix:auth-worker (pid=30842,uid=109): auth-worker<2>: Handling PASSV request
Dec 13 14:28:28 auth-worker(30843): Debug: conn unix:auth-worker (pid=30842,uid=109): auth-worker<2>: sql(manager@mydomain.com,45.125.66.22): Performing passdb lookup
Dec 13 14:28:28 auth-worker(30843): Debug: conn unix:auth-worker (pid=30842,uid=109): auth-worker<2>: sql(manager@mydomain.com,45.125.66.22): query: SELECT username AS user,password FROM mailbox WHERE username = 'manager@mydomain.com' AN
D active='1'
Dec 13 14:28:28 auth-worker(30843): Info: conn unix:auth-worker (pid=30842,uid=109): auth-worker<2>: sql(manager@mydomain.com,45.125.66.22): unknown user
Dec 13 14:28:28 auth-worker(30843): Debug: conn unix:auth-worker (pid=30842,uid=109): auth-worker<2>: sql(manager@mydomain.com,45.125.66.22): Finished passdb lookup
Dec 13 14:28:28 auth-worker(30843): Debug: conn unix:auth-worker (pid=30842,uid=109): auth-worker<2>: Finished
Dec 13 14:28:28 auth: Debug: sql(manager@mydomain.com,45.125.66.22): Finished passdb lookup
Dec 13 14:28:28 auth: Debug: auth(manager@mydomain.com,45.125.66.22): Auth request finished
Dec 13 14:28:28 auth: Debug: client passdb out: FAIL    1       user=eric@mydomain.com  original_user=eric
Dec 13 14:28:30 auth: Debug: client passdb out: FAIL    1       user=manager@mydomain.com       original_user=manager
`

**Comment by DavidGoodwin** [comment]
Yeah, the traditional response to this is ...

 * Ensure users have difficult to guess / not compromised passwords
 * Think of using something like fail2ban or crowdsec to block brute force bots (bear in mind it's fairly easy to do a slow distributed brute force of credentials, so over time I'd expect tools like crowdsec to become more common)
 * You could log where login requests come from - and e.g. if they're from a strange country, you trigger some sort of alert - see e.g. https://codepoets.co.uk/2022/postfix-postscreen-and-dns-blacklist-fun/ where dovecot can insert into a db table on successful login, you just then need to do something clever in your 'postlogin.sh' script ?
 
Alternatively, you could just ignore the logging and live with it ?
